sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/Hotspots/DoNotHardcodeCredentialsBase.cs

```diff
@@ -148,13 +151,30 @@ private IEnumerable<string> FindCredentialWords(string variableName, string vari
                 var matches = analyzer.passwordValuePattern.Matches(variableValue);
                 foreach (Match match in matches)
                 {
-                    credentialWordsFound.Add(match.Groups["password"].Value);
+                    if (!match.Success)
+                    {
+                        continue;
+                    }
+
+                    if (!IsValidParameter(match.Groups["suffix"].Value))
+                    {
+                        credentialWordsFound.Add(match.Groups["password"].Value);
+                    }
                 }
 
                 // Rule was initially implemented with everything lower (which is wrong) so we have to force lower
                 // before reporting to avoid new issues to appear on SQ/SC.
                 return credentialWordsFound.Select(x => x.ToLowerInvariant());
             }
+
+            private bool IsValidParameter(string suffix)
+            {
+                var candidateParameter = suffix
+                    .Split(new[] {';'}, StringSplitOptions.RemoveEmptyEntries)
+                    .FirstOrDefault();
```

Comment:
This connection string `password=;user=xxx` will produce `;user=xxx` suffix. Using `RemoveEmptyEntries` will remove the first empty split result (right?) and cause FP.
I suggest to add more connection strings to the tests.
I suggest to use `.Split(';').First().Trim()` and remove the check for null from next command.